### DISCUSSION TITLE
Testing

### DISCUSSION THREAD

**Discussion by imuhammadarsalan** [spatie/laravel-responsecache#387] [General]
Hello,

I installed the lib and configured it well, But I want to how can I test if the lib is caching the response? i made a simple project and it returns only the restapi response ..

Could you please guide me.

Thanks
Muhammad.

**Comment by rhnkyr** [comment]
You can something like this ( using pest ) :

```
test('it should hit the cache', function () {
    $cacheProfile = Mockery::mock(CacheProfile::class);
    $request = Request::create('https://api.test/api/user/1'); // Change with ur url
    $requestHasher = new DefaultHasher($cacheProfile);

    $cacheProfile->shouldReceive('useCacheNameSuffix')->andReturn('cacheProfileSuffix');

    $this->getJson(action([UserController::class], $this->user_id)) // $this->user_id comes from pest beforeEach()
        ->assertOk();

    $this->assertEquals('responsecache-xxxxxxxxxxxxxxxxx', $requestHasher->getHashFor($request)); // xxxxxxxxxxxxxxxxx hash of the cache key
});
```

**Comment by mateusjunges** [comment]
Hey, not sure if you want to add a PHPUnit test or just want to make sure that the response is being cached. In the second case, you can make a simple curl call using the `-v` param:

If your app is running at `localhost:8000`:

```curl
curl -s -I -X GET http://localhost:8000
```
It will return the response headers for you, and if you see a `laravel-responsecache: DATE`, than you can rest assured that you got a cached response:

![image](https://user-images.githubusercontent.com/19756164/211649426-5c37ed6a-6820-4723-8b4f-4ffd4363e65a.png)
